### PR TITLE
Router: fix handling of binpack defnSite

### PR DESCRIPTION
**Pull request by kitbellew**

### CODE REVIEW

**Review by tgodzik — APPROVED** [review]
LGTM!

**Comment by kitbellew** [comment]
thanks for the review. please do not merge it, I'd like to wait until #2749 is merged first, they overlap a little.

**Comment by kitbellew** [comment]
change my mind. let's merge this first.

**Pull request merged**
